File: GolangSolutions/Stack_Queue/M1441_用栈操作构建数组.go
package Stack_Queue

func buildArray(target []int, n int) (ans []string) {
	pre := 0
	for _, num := range target {
		for i := 0; i < num-pre-1; i++ {
			ans = append(ans, "Push", "Pop")
		}
		ans = append(ans, "Push")
		pre = num
	}

	return
}

File: LeetCode/src/Stack_Queue/M1441_用栈操作构建数组.java
package Stack_Queue;

import java.util.ArrayList;
import java.util.List;

/**
 * @author Hyperspace
 * @date 2022/10/15
 * <p>
 * 思路
 * 模拟
 */
public class M1441_用栈操作构建数组 {
    public List<String> buildArray(int[] target, int n) {
        int len = target.length;
        List<String> res = new ArrayList<>();
        int cnt = 0, num = 1;
        for (int i = 0; i < n && cnt != len; i++) {
            res.add("Push");
            if (target[cnt] == num) {
                cnt++;
            } else {
                res.add("Pop");
            }
            num++;
        }

        return res;
    }
}
